job_dsl/standard_job_checkmarx.groovy

```diff
@@ -0,0 +1,98 @@
+// Can't use env.FOO = {FOO} to transfer JJB vars to groovy
+// as this file won't be templated by JJB.
+// Alternative is to use parameters with JJB vars as the defaults.
+library "rpc-gating@${RPC_GATING_BRANCH}"
+common.globalWraps(){
+  common.standard_job_slave(env.SLAVE_TYPE) {
+    try {
+
+      stage("Configure Git"){
+        common.configure_git()
+      }
+
+      stage("Clone repo to scan"){
+        common.clone_with_pr_refs(
+          "repo",
+          SCAN_REPO_URL,
+          SCAN_REPO_BRANCH
+        )
+      }
+
+      stage("Checkmarx Scan"){
+        withCredentials([
+          string(
+            credentialsId: 'CHECKMARX_RE_TEAM_ID',
+            variable: 'groupId'
+          ),
+          string(
+            credentialsId: 'CHECKMARX_SERVER',
+            variable: 'serverUrl'
+          )
+        ]){
+          // Switch to scan repo dir to avoid sending the gating venv to checkmarx
+          dir("repo"){
+            presets = [
+              // values generated using the snippet generator
+              // ${jenkins}/pipeline-syntax/
+              // sample step > step: general build step
+              //    Build Step > Execute Checkmarx Scan
+              "default": "36",
+              "pci": "5"
+            ]
+            if (!presets.keySet().contains(SCAN_TYPE)){
+              throw new Exception("Invalid scan type: ${SCAN_TYPE}, should be default or pci")
+            }
+            step([$class: 'CxScanBuilder',
```

Comment:
Where did this configuration come from? Has it been reviewed by someone from the security team to ensure it is appropriate?

Reply:
It came from the [snippet generator](https://rpc.jenkins.cit.rackspace.net/pipeline-syntax/). It has not been reviewed. I'll ask @YasmineK to have a look.

Reply:
https://rackspace.slack.com/archives/C7B5ZBNUU/p1542127787008700

Reply:
@YasmineK had a look and [said](https://rackspace.slack.com/archives/C7B5ZBNUU/p1542130355009100) it looks ok ✅

Reply:
This conversation continued and as a result, I changed `default` to `all` for a broader scan.